templates/default/apache_site.conf.erb

```diff
@@ -51,8 +52,9 @@
 
     AllowOverride <%= @params['parse_htaccess'] ? "All" : "None" %>
 
-    Order allow,deny
-<%= @params['allow_from'] ? @params['allow_from'].map { |ip| "    Allow from #{ip}\n" } : "    Allow from all" %>
+<%= @params['allow_from'] || @params['basic_username'] ? "    Order deny,allow" : "    Order allow,deny" %>
+<%= @params['allow_from'] ? @params['allow_from'].map { |ip| "    Allow from #{ip}" }.join("\n") : '' %>
+<%= @params['allow_from'] || @params['basic_username'] ? "    Deny from all" : "    Allow from all" %>
```

Comment:
I see, a little confusion as Allow from all is set whether or not `Satisfy Any`

How about:

```
<% if @params['allow_from'] -%>
Order allow,deny
<%= @params['allow_from'].map { |ip| "    Allow from #{ip}" }.join("\n") %>
<%-end%>
```

This removes all reference to Order/allow/deny if no allow_from, and therefore no `Satisfy Any`